modules/mapbox/src/deck-utils.ts

```diff
@@ -50,7 +51,7 @@ export function getDeckInstance({
 
   let deckInstance: Deck;
 
-  if (!deck || deck.props.gl === gl) {
+  if (!deck || (deck.props.device instanceof WebGLDevice && deck.props.device.gl === gl)) {
```

Comment:
I assume the `instanceof` check here is just to keep Typescript happy? What happens if it is a `WebGPUDevice`? Would it maybe be safer to check for `WebGPUDevice` at the start of this function and exit early with a warning?